src/core/index.js

```diff
@@ -273,7 +273,8 @@ function extractAuthOptions(options) {
   // if responseType was not set and there is a redirectUrl, it defaults to code. Otherwise token.
   responseType = typeof responseType === 'string' ? responseType : redirectUrl ? 'code' : 'token';
   // now we set the default because we already did the validation
-  redirectUrl = redirectUrl || `${window.location.origin}${window.location.pathname}`;
+  redirectUrl =
+    redirectUrl || `${getOriginFromUrl(window.location.href)}${window.location.pathname}`;
```

Comment:
does this alternative of not using `window.location.origin` work in every scenario?

Reply:
yes, if anything, it's way more reliable, since works across all browsers.